### ISSUE TITLE
Getting duplicates in getPaged API call

### ISSUE THREAD

**Issue by revathy-kr** [pnp/pnpjs#1877]
### Category
- [ ] Bug

### Version

Please specify what version of the library you are using: [ 2.0.11]

Please specify what version(s) of SharePoint you are targeting: [1.11.0]

### Expected / Desired Behavior / Question
I have about 5000+ data in a list. When filtering along with orderBy as descending using a getPaged API call, I am getting duplicates in the response data. The API call looks like this:

_api/web/lists/getByTitle(%27Parties%27)/items?%24skiptoken=Paged%3dTRUE%26p_Title%3dTrailstone%2520UK%2520Ltd%252e%26p_ID%3d8485&%24select=PartyNo%2cTitle&%24expand=Author&%24orderby=Title+desc

**Comment by juliemturner** [comment]
So, you can see information like the ID that indicates they're duplicates vs list items with the same value in those two selected fields?

**Comment by revathy-kr** [comment]
Yes, the total response had items with the same ID being repeated.

**Comment by juliemturner** [comment]
Is Author (I'm assuming this is the internal Created By titled field) changed to have multiple values by any chance? Not that it should matter but I've seriously never seen anything like what you're talking about unless you use the search endpoint, then it's a very common thing.

**Comment by revathy-kr** [comment]
There are no multiple values in Created by field as it is a default field. The issue arises only when orderBy is descending.

**Comment by juliemturner** [comment]
Can you share the PnPjs code you're using the make the call?  The endpoint you've shared is the SharePoint REST API call, if you've tested that directly and that's where you're seeing the issue then there is nothing we can do as the library is only an abstraction layer for those REST api calls. If that is the case, you could post your questions in the sp-dev-docs issue list and hopefully you'll get more details there. If this is a PnPjs call and you're just sharing what the call is by using the browser network tools or fiddler can you validate that the result in those tools doesn't have duplicates, that would then indicate that somehow, we have a problem with the results that we are returning.

**Comment by koltyakov** [comment]
@revathy-kr,

Duplicates with pagination might appear when ordering by a field containing many identical values or empty values. This happening due a nature or indexing page token and SharePoint object model detecting wrong offset. A page skip token contains sort field and its last value in the collection.

To avoid duplication if it is appear an issue as a trick the sort condition might be amended to contain an extra field which would make pagination vector unique, e.g. adding the sort by Id in addition should help.
